modules/javafx.graphics/src/main/java/com/sun/glass/ui/win/WinSystemClipboard.java

```diff
@@ -252,7 +252,15 @@ protected final Object popFromSystem(String mimeFull) {
             if (TEXT_TYPE.equals(mime) || URI_TYPE.equals(mime)) {
                 try {
                     // JDK-8118474 - internal Windows data null terminated
-                    return new String(data, 0, data.length - 2, defaultCharset);
+                    // JDK-8281384 - buffer might be larger than data and null terminator not at the end
+                    int nullTerm = data.length - 2;
+                    for (int i = 0; i < data.length; i += 2) {
+                        if (data[i] == 0) {
```

Comment:
Since this is UTF-16, don't you need to check that both the even and odd bytes are 0?

```
                        if (data[i] == 0 && data[i+1] == 0) {
```

If you do this, you will want to validate that the length is even (if it isn't already ensured by `popBytes`).

Reply:
done and aborting loop now if there's only one byte left.

Reply:
Looks good now. I'll test and then finish my review.